realm/realm-library/src/main/java/io/realm/BaseRealm.java

```diff
@@ -517,6 +505,7 @@ public RealmSchema getSchema() {
 
     // Used by RealmList/RealmResults
     // Invariant: if dynamicClassName != null -> clazz == DynamicRealmObject
+    // TODO: Remove this after RealmList is backed by OS Results.
```

Comment:
Like the `RealmResults`, `RealmList` could also use OS `Results` (java internal `Collection`) instead of `LinkView`. The benefit is then we can directly add listeners on `RealmList` instead of call `where` to create a `RealmResults` frist. If we do that, this function is not needed any more since OS `Results::get()` can return a `row` pointer directly to create a java `Row` object from that. Just like how we create Row from RealmResults in this PR.